src/compute/src/render/reduce.rs

```diff
@@ -2171,6 +2178,49 @@ impl Columnation for Accum {
     type InnerRegion = CopyRegion<Self>;
 }
 
+/// Helper to merge pairs of datum iterators into a row or split a datum iterator
+/// into two rows, given the arity of the first component.
+#[derive(Clone, Copy, Debug)]
+struct Pairer {
+    split_arity: usize,
+}
+
+impl Pairer {
+    /// Creates a pairer with knowledge of the arity of first component in the pair.
+    fn new(split_arity: usize) -> Self {
+        Self { split_arity }
+    }
+
+    /// Merges a pair of datum iterators creating a `Row` instance.
+    fn merge<'a, I1, I2>(&self, pair: (I1, I2)) -> Row
```

Comment:
I wouldn't specify the parameters as a pair because in all call sites we have to construct the pair, there is no situation where we can pass it through from the outside I think.

Reply:
Sure thing, revised.